ucp/_libs/ucx_api.pyx

```diff
@@ -1443,3 +1561,196 @@ def stream_recv_nb(
+        whenever the send operation will be completed. In other words, the completion
+        of a message can be signaled by the return code or the call-back.
+
+        Note
+        ----
+        The user should not modify any part of the buffer after this operation is
+        called, until the operation completes.
+
+        Parameters
+        ----------
+        ep: UCXEndpoint
+            The destination endpoint
+        buffer: Array
+            An ``Array`` wrapping a user-provided array-like object
+        nbytes: int
+            Size of the buffer to use. Must be equal or less than the size of buffer
+        cb_func: callable
+            The call-back function, which must accept `request` and `exception` as the
+            first two arguments.
+        cb_args: tuple, optional
+            Extra arguments to the call-back function
+        cb_kwargs: dict, optional
+            Extra keyword arguments to the call-back function
+        name: str, optional
+            Descriptive name of the operation
+        """
+        if cb_args is None:
+            cb_args = ()
+        if cb_kwargs is None:
+            cb_kwargs = {}
+        if name is None:
+            name = "am_send_nb"
+        if Feature.AM not in ep.worker._context._feature_flags:
+            raise ValueError("UCXContext must be created with `Feature.AM`")
+        if buffer.cuda and not ep.worker._context.cuda_support:
+            raise ValueError(
+                "UCX is not configured with CUDA support, please add "
+                "`cuda_copy` and/or `cuda_ipc` to the UCX_TLS environment"
+                "variable and that the ucx-proc=*=gpu package is "
+                "installed. See "
+                "https://ucx-py.readthedocs.io/en/latest/install.html for "
+                "more information."
+            )
+        if not buffer._contiguous():
+            raise ValueError("Array must be C or F contiguous")
+
+        cdef ucp_request_param_t params
+        params.op_attr_mask = (
+            UCP_OP_ATTR_FIELD_CALLBACK |
+            UCP_OP_ATTR_FIELD_USER_DATA |
+            UCP_OP_ATTR_FIELD_FLAGS
+        )
+        params.cb.send = <ucp_send_nbx_callback_t>_send_nbx_callback
+        params.flags = UCP_AM_SEND_FLAG_REPLY
+        params.user_data = <void*>buffer.ptr
+
+        cdef ucs_status_ptr_t status = ucp_am_send_nbx(
+            ep._handle,
+            0,
+            NULL,
+            0,
+            <void*>buffer.ptr,
+            nbytes,
+            &params,
+        )
+        return _handle_status(
+            status, nbytes, cb_func, cb_args, cb_kwargs, name, ep._inflight_msgs
+        )
+
+    def am_recv_nb(
+        UCXEndpoint ep,
+        cb_func,
+        tuple cb_args=None,
+        dict cb_kwargs=None,
+        str name=None,
+    ):
+        """ This routine receives a message on a worker with the active message API.
+
+        TODO
```

Comment:
Like with `am_send_nbx()`, it would be good with the UCX doc here.

Reply:
I left as TODO for the #707 , as I'm not confident yet as what to write or how it will look as it's still WIP.